File: problem/bool-and-or/6/main.cs
// main.cs: main file for solutions to the bool-not exercise
//
// This is to be linked to the submitted file.
// It processes values from standard input then from the "in.txt" file.
//
//
// Copyright (C) 2020-2021  Rudy Matela
// All rights reserved.

using System;

class TheNewMain
{
	static void Main()
	{
		string line;

		// processes standard input
		while ((line = Console.ReadLine()) != null) {
			Solve1(line);
		}

		// processes additional inputs from in.txt
		var file = new System.IO.StreamReader("in.txt");
		while ((line = file.ReadLine()) != null) {
			Solve1(line);
		}
		file.Close();
	}

	static void Solve1(string line) {
		string[] inputs = line.Split(null);
		bool p = inputs[0] == "true";
		bool q = inputs[2] == "true";
		if (inputs[1] == "and")
			Console.WriteLine(Program.Conjunction(p,q) ? "true" : "false");
		else
			Console.WriteLine(Program.Disjunction(p,q) ? "true" : "false");
	}
}

File: problem/bool-and-or/6/main.js
// main.js: main file for solutions to the triple exercise
//
// It processes values from standard input then from the "in.txt" file.
// This is appended to the submitted JavaScript program and tests the triple function.
//
// The submitted file processes the standard input and this processes the
// "in.txt" file.
//
// Copyright (C) 2020-2021  Rudy Matela
// All rights reserved.

var fs = require('fs').createReadStream('in.txt');
var rl = require('readline').createInterface({input: fs, output: process.stdout});

rl.on('line', function(line){
  words = line.split(/ /);
  p = words[0] == "true";
  o = words[1] == "and" ? conjunction : disjunction;
  q = words[2] == "true";
  console.log(o(p,q));
});
